js/data/data_source.d.ts

```diff
@@ -434,6 +431,11 @@ export default class DataSource
  * @docid
  * @type Store|DataSource|DataSourceOptions|string|Array<any>
  * */
-export type DataSourceLike<TItem, TKey = any> = string | Array<TItem> | Store<TItem, any, TKey> | Options<any, TItem, any, any, TKey> | DataSource<TItem, any, TKey>;
+export type DataSourceLike<TItem, TKey = any> =
+    string |
+    Array<TItem> |
+    Store<TItem, TKey> |
+    Options<any, TItem, any, TKey> |
```

Comment:
Since the TMappedItem is the second generic arg, you should change order here:
```suggestion
    Options<any, any, TItem, TKey> |
```

Reply:
Thanks, fixed